llama_stack/providers/inline/datasetio/localfs/datasetio.py

```diff
@@ -172,7 +172,7 @@ async def append_rows(self, dataset_id: str, rows: List[Dict[str, Any]]) -> None
         new_rows_df = dataset_impl._validate_dataset_schema(new_rows_df)
         dataset_impl.df = pandas.concat([dataset_impl.df, new_rows_df], ignore_index=True)
 
-        url = str(dataset_info.dataset_def.url)
+        url = str(dataset_info.dataset_def.url.uri)
```

Comment:
Can you share more details on how you are constructing the client calls?

https://github.com/meta-llama/llama-stack/blob/af396e3809bff1bad1b82a29c89da4048e8605bd/llama_stack/apis/common/content_types.py#L16-L18

URL should contain a uri field.

Reply:
This is how I call
```
url='file:///home/xxx/yyy/zzz/data.csv'
url=drp.URL(uri=url)
client.datasets.register(dataset_id=dsid, url=url, provider_id=provider_id,dataset_schema=schema)
```

later in the line I fixed:
`str(dataset_info.dataset_def.url)` evaluates to the string "uri='file:///home/xxx/yyy/zzz/data.csv'" while the fix `str(dataset_info.dataset_def.url.uri)` evaluates to the string "file:///home/jsalomon/src/llama_stack/data.csv"

Reply:
> str(dataset_info.dataset_def.url) evaluates to "uri='file:///home/xxx/yyy/zzz/data.csv'"
> str(dataset_info.dataset_def.url.uri) evaluates to "file:///home/jsalomon/src/llama_stack/data.csv"

Yes, that is expected? Why is the change necessary?

Reply:
@yanxi0830 see the comment below explaining what happens.
The change is necessary for 2 reasons:

- If you pass a data file to the llama server, the code will try to load from another file, meaning it will always start from an empty file.
- It is impossible to debug and analyze when you expect a file in some place but is exists in a completely different place - this makes troubleshooting a nightmare
- no schema of URI but 'file' is supported since the filename can't be parsed as a normal URI and the default is file, the means that data uris will never work

Reply:
Please handle this - it is a long time for a one-liner to fix a clear bug.